bigbluebutton-html5/imports/ui/components/user-list/service.js

```diff
@@ -204,13 +204,54 @@ const getOpenChats = (chatID) => {
     .sort(sortChats);
 };
 
+const getAvailableActions = (currentUser, user, router, isBreakoutRoom) => {
+  const hasAuthority = currentUser.isModerator || user.isCurrent;
+  const allowedToChatPrivately = !user.isCurrent;
+  const allowedToMuteAudio = hasAuthority && user.isVoiceUser && user.isMuted;
+  const allowedToUnmuteAudio = hasAuthority && user.isVoiceUser && !user.isMuted;
+  const allowedToResetStatus = hasAuthority && user.emoji.status !== 'none';
+
+  // if currentUser is a moderator, allow kicking other users
+  const allowedToKick = currentUser.isModerator && !user.isCurrent && !isBreakoutRoom;
+
+  const allowedToSetPresenter = currentUser.isModerator && !user.isPresenter;
+
+  const allowedToPromote = currentUser.isModerator && !user.isCurrent && !user.isModerator;
+  const allowedToDemote = currentUser.isModerator && !user.isCurrent && user.isModerator;
+
+  return {
+    allowedToChatPrivately,
+    allowedToMuteAudio,
+    allowedToUnmuteAudio,
+    allowedToResetStatus,
+    allowedToKick,
+    allowedToSetPresenter,
+    allowedToPromote,
+    allowedToDemote,
+  };
+};
+
 const getCurrentUser = () => {
   const currentUserId = Auth.userID;
   const currentUser = Users.findOne({ userId: currentUserId });
 
   return (currentUser) ? mapUser(currentUser) : null;
 };
 
+const normalizeEmojiName = (emoji) => {
+  const emojisNormalized = {
```

Comment:
Is this normalization aiming to match status with icon name? It defeats the purpose of extracting the list of emoji strings in a config file

Reply:
The purpose of this function, is for backward compatibility, we might remove this